### PR TITLE
feat(nimbus): Include stdout and stderr in manifesttool summaries

### LINKED ISSUE #10678: Include stdout and stderr in manifesttool summaries
When a fetch fails due to a subprocess error (e.g., see https://github.com/mozilla/experimenter/pull/10675#issuecomment-2102878797), the stdout and stderr of the command are not included in the summary. Instead, they are only present in the log in CI. We should include this in our summaries to make debugging easier, especially so we can ignore spurious network failures.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-4523)

### PR DESCRIPTION
**Pull request by brennie**
Because:

- manifesttool does not include stdout and stderr of subcommands in its summaries;
- the most common failure so far has been intermittent network issues in nimbus-cli; and
- we currently have to check the CI logs to find out why commands fail

This commit:

- includes stdout and stderr of subcommands in the summaries.

Fixes #10678

### CODE REVIEW

File: experimenter/manifesttool/tests/test_fetch.py [hunk 1]
```diff
@@ -1011,30 +1025,74 @@ def test_summarize_results(self):
...
+        summary = re.sub(
+            r"""File "/experimenter/manifesttool/tests/test_fetch.py", line \d+, in test_summarize_results""",
+            """File "/experimenter/manifesttool/tests/test_fetch.py", line 0000, in test_summarize_results""",
+            summary,
         )
+
+<<<<<<< Updated upstream
+
+=======
+>>>>>>> Stashed changes
```
Comment (jaredlockhart) [inline experimenter/manifesttool/tests/test_fetch.py:1058]: What's going on here?

**Comment by brennie** [comment]
Here is an example of what the new summaries look like:

```
SUMMARY:

FAILURES:

bogus at main (6b2b931f6e57e93af7ec6ad8b6e6e50a376a26fc) version None
Traceback (most recent call last):
  File "/experimenter/manifesttool/fetch.py", line 80, in fetch_fml_app
    channels = nimbus_cli.get_channels(app_config, fml_path, ref.target)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/experimenter/manifesttool/nimbus_cli.py", line 33, in get_channels
    output = nimbus_cli(
             ^^^^^^^^^^^
  File "/experimenter/manifesttool/nimbus_cli.py", line 27, in nimbus_cli
    raise e
  File "/experimenter/manifesttool/nimbus_cli.py", line 19, in nimbus_cli
    return subprocess.check_output(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/application-services/bin/nimbus-cli', 'fml', '--', 'channels', '--json', '--ref', '6b2b931f6e57e93af7ec6ad8b6e6e50a376a26fc', '@mozilla-mobile/firefox-ios/bogus.yml']' returned non-zero exit status 1.

stderr:
-----
Error: Problem with https://raw.githubusercontent.com/mozilla-mobile/firefox-ios/6b2b931f6e57e93af7ec6ad8b6e6e50a376a26fc/bogus.yml: YAML Error: unknown field `404` at line 1 column 4
-----
```

**Review by jaredlockhart — COMMENTED** [review]
Is there really no good way to do this in bash space or something?  Without having to explicitly manage all this in Python space?

File: experimenter/manifesttool/tests/test_fetch.py (hunk 1 above)
Comment (brennie) [reply experimenter/manifesttool/tests/test_fetch.py]: A bad rebase :)

**Review by jaredlockhart — APPROVED** [review]
As in just tack on `&> filename` whenever you invoke any of the external commands and then dump that out later.  But if this is easier then let's do it 👍

**Comment by brennie** [comment]
> As in just tack on `&> filename` whenever you invoke any of the external commands and then dump that out later. 

This would be more work since Python's subprocess module can just collect the output (which is what we're doing with `subprocess.check_output` and `stderr=subprocess.PIPE`). WE'd have to keep track of all of that and then also read files ourselves, which seems like more work.

**Pull request merged**